clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordsTest.java

```diff
@@ -35,28 +35,83 @@ public class ConsumerRecordsTest {
+        Iterator<ConsumerRecord<Integer, String>> iterator = records.iterator();
+
+        for (int count = 0; iterator.hasNext(); count++) {
+            if (count % emptyPartitionInterval != 0) {
+                int offset = 0;
+                for (; offset < recordSize; offset++) {
+                    ConsumerRecord<Integer, String> record = iterator.next();
+                    assertEquals(count, record.partition());
+                    assertEquals(topic, record.topic());
+                    assertEquals(offset, record.offset());
+                }
+                assertEquals(recordSize, offset);
+            }
+        }
+    }
 
+    @Test
+    public void testRecordsWithNullTopic() {
+        String nullTopic = null;
         Map<TopicPartition, List<ConsumerRecord<Integer, String>>> records = new LinkedHashMap<>();
+        ConsumerRecords<Integer, String> consumerRecords = new ConsumerRecords<>(records);
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> consumerRecords.records(nullTopic));
+        assertEquals("Topic must be non-null.", exception.getMessage());
+    }
 
-        String topic = "topic";
-        records.put(new TopicPartition(topic, 0), new ArrayList<>());
-        ConsumerRecord<Integer, String> record1 = new ConsumerRecord<>(topic, 1, 0, 0L, TimestampType.CREATE_TIME,
-            0, 0, 1, "value1", new RecordHeaders(), Optional.empty());
-        ConsumerRecord<Integer, String> record2 = new ConsumerRecord<>(topic, 1, 1, 0L, TimestampType.CREATE_TIME,
-            0, 0, 2, "value2", new RecordHeaders(), Optional.empty());
-        records.put(new TopicPartition(topic, 1), Arrays.asList(record1, record2));
-        records.put(new TopicPartition(topic, 2), new ArrayList<>());
+    @Test
+    public void testRecords() {
+        String[] topics = {"topic1", "topic2", "topic3", "topic4"};
+        int recordSize = 3;
+        int partitionSize = 10;
+        int emptyPartitionInterval = 3;
+        ConsumerRecords<Integer, String> consumerRecords = buildTopicTestRecords(recordSize, partitionSize, emptyPartitionInterval, topics);
 
-        ConsumerRecords<Integer, String> consumerRecords = new ConsumerRecords<>(records);
-        Iterator<ConsumerRecord<Integer, String>> iter = consumerRecords.iterator();
-
-        int c = 0;
-        for (; iter.hasNext(); c++) {
-            ConsumerRecord<Integer, String> record = iter.next();
-            assertEquals(1, record.partition());
-            assertEquals(topic, record.topic());
-            assertEquals(c, record.offset());
+        for (String topic : topics) {
+            Iterable<ConsumerRecord<Integer, String>> records = consumerRecords.records(topic);
+            Iterator<ConsumerRecord<Integer, String>> iterator = records.iterator();
+
+            for (int count = 0; iterator.hasNext(); count++) {
```

Comment:
Could you verify the number of records from the iterable/iterator?

Reply:
Sure, I will add the verification of it.